
# HG changeset patch
# User sgehwolf
# Date 1542877918 -3600
# Node ID 7088cfa71363656579b585f14d55ad45c654a7a0
# Parent  4f45c682eab0ef38fc3b81681589590479994c87
8214105: [TESTBUG] Bit test is always false in serviceability/jvmti/StartPhase/AllowedFunctions
Reviewed-by: dcubed, dholmes, jcbeyler, gadams, sspitsyn
Contributed-by: Simon Tooke <stooke@redhat.com>

diff -r 4f45c682eab0 -r 7088cfa71363 test/hotspot/jtreg/serviceability/jvmti/StartPhase/AllowedFunctions/libAllowedFunctions.c
--- a/test/hotspot/jtreg/serviceability/jvmti/StartPhase/AllowedFunctions/libAllowedFunctions.c	Thu Nov 22 10:01:38 2018 +0100
+++ b/test/hotspot/jtreg/serviceability/jvmti/StartPhase/AllowedFunctions/libAllowedFunctions.c	Thu Nov 22 10:11:58 2018 +0100
@@ -110,8 +110,8 @@
     // This function is only used in a ClassPrepare event context
     if ((status & JVMTI_CLASS_STATUS_VERIFIED)    == 0 ||
         (status & JVMTI_CLASS_STATUS_PREPARED)    == 0 ||
-        (status & JVMTI_CLASS_STATUS_INITIALIZED) == 1 ||
-        (status & JVMTI_CLASS_STATUS_ERROR)       == 1) {
+        (status & JVMTI_CLASS_STATUS_INITIALIZED) != 0 ||
+        (status & JVMTI_CLASS_STATUS_ERROR)       != 0) {
         printf("  ## Error: unexpected class status: 0x%08x\n", status);
     }
     printf("    Class status: 0x%08x\n", status);

